lib/instrumentation/winston.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const createFormatter = require('@newrelic/winston-enricher/lib/createFormatter')
+const API = require('../../api')
+
+module.exports = function instrument(agent, winston, _, shim) {
+  const config = agent.config
+  if (
+    !(
+      config.application_logging &&
+      config.application_logging.enabled &&
+      config.application_logging.local_decorating &&
+      config.application_logging.local_decorating.enabled
+    )
+  ) {
+    shim.logger.debug('Application logging not enabled. Not instrumenting winston...')
+    return // do not instrument logs if logging is not enabled
+  }
+
+  // create an instrumented formatter to enrich and report logs
+  const api = new API(agent)
+  api.shim = shim
+  const instrumentedFormatter = createFormatter(api, winston)
+
+  // wrap logger creation to combine or insert our formatter
+  shim.wrap(winston, 'createLogger', function wrapCreate(shim, createLogger) {
+    return function createWrappedLogger(opts = {}) {
+      // combine top-level formatter with our own
+      if ('format' in opts) {
+        opts.format = winston.format.combine(opts.format, instrumentedFormatter())
+      } else {
```

Comment:
you may be more familiar than I but shouldn't we check for transports in this if/else statement as well? it seems like we're adding a formatter always if we didn't specify a format.  So in the case where you define a transport we are adding a top level format with just the winston log enricher and then within the stream if you're specifying a format but not when you aren't.  Maybe we can talk through this case IRL

Reply:
I've adjusted the code to address this concern. Thanks for raising it!